main.go

```diff
@@ -202,13 +202,18 @@ func main() {
 	prometheus.MustRegister(collector.NewNodes(logger, httpClient, esURL, *esAllNodes, *esNode))
 
 	if *esExportIndices || *esExportShards {
-		prometheus.MustRegister(collector.NewShards(logger, httpClient, esURL))
+		sC := collector.NewShards(logger, httpClient, esURL)
+		prometheus.MustRegister(sC)
 		iC := collector.NewIndices(logger, httpClient, esURL, *esExportShards, *esExportIndexAliases)
 		prometheus.MustRegister(iC)
-		if registerErr := clusterInfoRetriever.RegisterConsumer(iC); registerErr != nil {
+		if clusterInfoRetriever.RegisterConsumer(iC) != nil {
```

Comment:
This isn't idiomatic go. 
```suggestion
		if registerErr := clusterInfoRetriever.RegisterConsumer(iC); registerErr != nil {
```